src/chainstate/stacks/boot/pox-2.clar

```diff
@@ -966,21 +997,34 @@
                     (pox-addr { version: (buff 1), hashbytes: (buff 20) })
                     (extend-count uint))
     (let ((stacker-info (stx-account stacker))
+          (stacker-state (get-stacker-info stacker))
           (amount-ustx (get locked stacker-info))
-          (unlock-height (get unlock-height stacker-info)))
+          (unlock-height (get unlock-height stacker-info))
+          (unlock-in-cycle (burn-height-to-reward-cycle unlock-height))
+          ;; if the account unlocks *during* this cycle (should only occur during testing),
```

Comment:
"[This] should only occur during testing" sounds like someone's famous last words :)  Why would it be the case that the account unlocks during this cycle?  Wouldn't the delegate be able to extend their lock in the same cycle in which they unlock?

Reply:
It's if `v1_unlock_ht` is set to the middle of a cycle rather than a boundary. It could occur if that's how the 2.1 fork specifies that value, but even in that case, this code will handle the case smoothly by setting the `first-extended-cycle` to the next cycle.

Reply:
If we're certain that `v1_unlock_ht` will be set on a reward cycle boundary, then should there be a runtime assert that this is the case if we're on mainnet?